SMLHelper/MonoBehaviours/EntitySpawner.cs

```diff
@@ -24,24 +26,43 @@ IEnumerator SpawnAsync()
             var task = new TaskResult<GameObject>();
             yield return GetPrefabAsync(task);
 
-            if (task.Get() is null)
+            var prefab = task.Get();
+
+            if (prefab == null)
             {
                 Logger.Error($"no prefab found for {stringToLog}; process for Coordinated Spawn canceled.");
                 Destroy(gameObject);
             }
 
-            var obj = UWE.Utils.InstantiateDeactivated(task.Get(), spawnInfo.spawnPosition, spawnInfo.rotation);
+
+            var obj = Utils.InstantiateDeactivated(prefab, spawnInfo.spawnPosition, spawnInfo.rotation);
+
+            var lwe = obj.GetComponent<LargeWorldEntity>();
+
+            // non-global objects cannot be spawned in unloaded terrain so we need to wait
+            if (lwe == null || lwe.cellLevel is not (LargeWorldEntity.CellLevel.Batch or LargeWorldEntity.CellLevel.Global))
+            {
+                var lws = LargeWorldStreamer.main;
+                while (!lws.IsReady()) // first we make sure the world streamer is initialized
+                {
+                    yield return new WaitForSecondsRealtime(2f);
+                }
```

Comment:
Is there something we can wait for that isn't just an arbitrary 2 second timer in a while loop? Would `new WaitUntil(() => lws.IsReady()` not be more suitable, since it seems like you're just trying to wait until it is ready?

```suggestion
                yield return new WaitUntil(() => lws.IsReady()); // first we make sure the world streamer is initialized
```

Reply:
Right, i wanted to use a `WaitUntil` for that one as well, but i thought checking for it on each frame wasn't necessary so i did that 2 seconds thing.

Reply:
How heavy is the `IsReady()` method call? If not very heavy, it's probably worthwhile. An arbitrary 2 second wait seems fragile to me.

Reply:
yeah, it's not very heavy, it just checks for an `init` boolean field somewhere else, we can change it.